### ISSUE TITLE
no-undef misses JSX references

### ISSUE THREAD

**Issue by yannickcr** [eslint/eslint#1676]
The no-undef rules misses the references to undefined variables in JSX.

Examples:

```jsx
React.render(<UnknownApp />);
```
Should report `'UnknownApp' is not defined`

```jsx
React.render(<div className={unknownClass} />);
```
Should report `'unknownClass' is not defined`

**Comment by nzakas** [comment]
This is going to be really tough without escope support. Not sure how to deal with this.

**Comment by yannickcr** [comment]
I tried something, see #1683

Hope I did not miss some cases.

**Comment by fab1an** [comment]
I still have this problem.

**Comment by ktoh** [comment]
+1

**Comment by fab1an** [comment]
It works using this plugin: https://www.npmjs.com/package/eslint-plugin-react
